subsys/shell/shell_cmds.c

```diff
@@ -198,143 +194,107 @@ static int cmd_clear(const struct shell *shell, size_t argc, char **argv)
 
 static int cmd_shell(const struct shell *shell, size_t argc, char **argv)
 {
-	int ret = shell_cmd_precheck(shell, (argc == 2), NULL, 0);
-
-	if (ret) {
-		return ret;
-	}
-
-	shell_fprintf(shell, SHELL_ERROR, "%s:%s%s\n", argv[0],
-		      SHELL_MSG_UNKNOWN_PARAMETER, argv[1]);
+	shell_error(shell, "%s:%s%s", argv[0], SHELL_MSG_UNKNOWN_PARAMETER,
+		    argv[1]);
 
 	return -EINVAL;
 }
 
 static int cmd_bacskpace_mode_backspace(const struct shell *shell, size_t argc,
 					char **argv)
 {
+	ARG_UNUSED(argc);
 	ARG_UNUSED(argv);
 
-	int ret = shell_cmd_precheck(shell, (argc == 1), NULL, 0);
+	shell->ctx->internal.flags.mode_delete = 0;
 
-	if (ret == 0) {
-		shell->ctx->internal.flags.mode_delete = 0;
-	}
-
-	return ret;
+	return 0;
 }
 
 static int cmd_bacskpace_mode_delete(const struct shell *shell, size_t argc,
 				      char **argv)
 {
+	ARG_UNUSED(argc);
 	ARG_UNUSED(argv);
 
-	int ret = shell_cmd_precheck(shell, (argc == 1), NULL, 0);
+	shell->ctx->internal.flags.mode_delete = 1;
 
-	if (ret == 0) {
-		shell->ctx->internal.flags.mode_delete = 1;
-	}
-
-	return ret;
+	return 0;
 }
 
 static int cmd_bacskpace_mode(const struct shell *shell, size_t argc,
```

Comment:
this function seems to be redundant now. handler could be null.

Reply:
Will fix.